File: frontend/src/services/PublicationService.js
import axios from "../custom-axios/axios"

const PublicationService={
    getAllPublicationsByCategory: ()=>{
        return axios.get('/publications')
    },

    addPublicationSuggestion: (name, authorsNames, url, description)=>{
        return axios.post('/publications/addSuggestion',{
            "name":name,
            "authorsNames":authorsNames,
            "url":url,
            "description":description
        })
    }
}

export default PublicationService;

File: backend/src/main/java/com/example/backend/service/PublicationService.java
package com.example.backend.service;

import com.example.backend.model.Publication;

import java.math.BigInteger;
import java.util.List;
import java.util.Optional;

public interface PublicationService {
    List<Publication> getAllPublications();
    void addNewPublication(String publicationId, BigInteger numberOfEntitiesFromPublication,
                           String publicationFromCategory, String publicationLabel);
    void addPublicationsForCategory(String category);
    void addNewPublicationSuggestion(String name, String authorsNames, String url, String description);
}
